questions-encodings.mjs

```diff
@@ -1,30 +1,45 @@
 // server side script called by GitHub Actions to periodically generate encodings stampy.ai wiki questions
 
-// npm install node-fetch @tensorflow/tfjs-node @tensorflow-models/universal-sentence-encoder
+// npm install process node-fetch @tensorflow/tfjs-node @tensorflow-models/universal-sentence-encoder
 import * as tf from '@tensorflow/tfjs-node'
 import * as use from '@tensorflow-models/universal-sentence-encoder'
 import fetch from 'node-fetch'
+import exit from 'process'
 import * as fs from 'fs'
 
-//const filepath = '~/public/assets/'
-const filename = 'stampy-questions-encodings.json'
+const filepath = 'public/assets/'
+const filename = filepath + 'stampy-questions-encodings.json'
+
+let prevNumQs = 0
+if (fs.existsSync(filename)) {
+  const p = JSON.parse(fs.readFileSync(filename, 'utf8'))
+  prevNumQs = p.numQs
+  console.log(`${prevNumQs} questions from in previous generated encodings.`)  
+}
 
 const url =
   'https://stampy.ai/w/api.php?action=ask&query=[[Canonically%20answered%20questions]]|format%3Dplainlist|%3FCanonicalQuestions&format=json'
 const r = await (await fetch(url)).json()
 const response = r.query.results['Canonically answered questions'].printouts.CanonicalQuestions
-console.log(`Fetched ${response.length} questions from the stampy.ai wiki.`)
+console.log(`${response.length} questions fetched from the stampy.ai wiki.`)
 
 const questions = response.map(({fulltext}) => fulltext)
 const questionsNormalized = questions.map((question) => question.toLowerCase())
+const numQs = questions.length
+
+if (prevNumQs == numQs) {
+  console.log(`Number of canonically answered questions has not changed. No new encodings needed.`)
+  // print or return something to let calling script know nothing changed, not to commit?
+  process.exit(0)
```

Comment:
0 is a success code, let's try to exit 1 to indicate error so github action would not continue:
```suggestion
  process.exit(1)
```

Reply:
aah, that worked for skipping commit & push, but failing this step made whole action fail too:

<img width="411" alt="Screenshot 2022-07-01 at 20 18 35" src="https://user-images.githubusercontent.com/1087670/176957002-6a035e9e-5754-46e0-9dac-f49808416d88.png">

I tried to experiment, but didn't find anything better right now => #41 to improve